internal/transport/grpchttp2/framer.go

```diff
@@ -19,7 +19,12 @@
 // Package grpchttp2 defines HTTP/2 types and a framer API and implementation.
 package grpchttp2
 
-import "golang.org/x/net/http2/hpack"
+import (
+	"golang.org/x/net/http2/hpack"
+	"google.golang.org/grpc/mem"
+)
+
+const initHeaderTableSize = 4096 // Default HTTP/2 header table size.
```

Comment:
We do have a server option to set the header table size. See: https://pkg.go.dev/google.golang.org/grpc#HeaderTableSize

How does this interact with that option?

Reply:
Looks like this is what is happening:
- We have the const as you mention [here](https://github.com/grpc/grpc-go/blob/e524655becd8d4c7ba9e8687faef456e495e341e/internal/transport/http_util.go#L46), and this is used when creating the hpack decoder and this specifies the initial table size
- The server option I specified results in the server sending out a setting in the settings frame: https://github.com/grpc/grpc-go/blob/e524655becd8d4c7ba9e8687faef456e495e341e/internal/transport/http2_server.go#L198-L203
- When a SETTINGS frame is received by either the client or the server, they enqueue an `incomingSettings` control frame on the `controlbuf`. This results in the loopy writer handling this particular setting [here](https://github.com/grpc/grpc-go/blob/e524655becd8d4c7ba9e8687faef456e495e341e/internal/transport/controlbuf.go#L906), by changing the size of the table.

Reply:
So, what you have currently should be good enough to set the initial table size we just need to double check that the loopy writer will update the table size in the new framer as well (based on the same setting received in a SETTINGS frame).